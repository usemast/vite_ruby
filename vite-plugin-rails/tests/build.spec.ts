import { resolve } from 'path'
import { readFileSync } from 'fs'
import { beforeAll, describe, test, expect } from 'vitest'
import execa from 'execa'
import glob from 'fast-glob'

const projectRoot = resolve(__dirname, '../')
const exampleDir = `${projectRoot}/example`

describe('config', () => {
  beforeAll(async () => {
    await execa('npm', ['run', 'build'], { stdio: process.env.DEBUG ? 'inherit' : undefined, cwd: exampleDir })
  }, 60000)

  test('generated files', async () => {
    const outDir = `${exampleDir}/public/vite`
    const files = await glob('**/*', { cwd: outDir, onlyFiles: true })
    expect(files.sort()).toEqual(expect.arrayContaining([
      'assets/app-HiPXl-_u.css',
      'assets/app-HiPXl-_u.css.br',
      'assets/app-HiPXl-_u.css.gz',
      'assets/external-TUcZQQlC.js',
      'assets/external-TUcZQQlC.js.br',
      'assets/external-TUcZQQlC.js.gz',
      'assets/external-TUcZQQlC.js.map',
      'assets/index-ChIRGsHu.js',
      'assets/index-ChIRGsHu.js.br',
      'assets/index-ChIRGsHu.js.gz',
      'assets/index-ChIRGsHu.js.map',
      'assets/logo-DbqOB_Bf.svg',
      'assets/logo-Yf1y9V5e.png',
      'assets/logo-j5j6qipL.png',
      'assets/main-E96L_vrM.js',
      'assets/main-E96L_vrM.js.br',
      'assets/main-E96L_vrM.js.gz',
      'assets/main-E96L_vrM.js.map',
      'assets/sassy-NCcoc604.css',
      'assets/sassy-NCcoc604.css.br',
      'assets/sassy-NCcoc604.css.gz',
      'assets/theme-gVQjH5fM.css',
      'assets/theme-gVQjH5fM.css.br',
      'assets/theme-gVQjH5fM.css.gz',
      'assets/vue-Mdlordc1.css',
      'assets/vue-Mdlordc1.css.br',
      'assets/vue-Mdlordc1.css.gz',
      'assets/vue-eqttolrD.js',
      'assets/vue-eqttolrD.js.br',
      'assets/vue-eqttolrD.js.gz',
      'assets/vue-eqttolrD.js.map',
      'index.html',
      'index.html.br',
      'index.html.gz',
    ]))

    const parseManifest = (path: string) => JSON.parse(readFileSync(`${outDir}/.vite/${path}`, 'utf-8'))
    const manifest = parseManifest('manifest.json')
    const manifestAssets = parseManifest('manifest-assets.json')

    expect({ ...manifest, ...manifestAssets }).toMatchSnapshot()
  })
})
