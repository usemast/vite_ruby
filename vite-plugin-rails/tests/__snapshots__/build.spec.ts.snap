// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`config > generated files 1`] = `
{
  "../assets/external.js": {
    "file": "assets/external-TUcZQQlC.js",
    "integrity": "sha384-afpxQsY7Zl0vHthljoHJFlj4eX3fvxujTRCy532FF3prt0UsynDiMGZaI9HOENvv",
    "isEntry": true,
    "src": "../assets/external.js",
  },
  "../assets/logo.png": {
    "file": "assets/logo-j5j6qipL.png",
    "integrity": "sha384-IGw3tfo4Xt0lp6sO9t5mYtILwfBxDFR8sRCtroxGjhE3r9UxNYgtimkCNzsJU9AP",
    "src": "app/assets/logo.png",
  },
  "../assets/theme.css": {
    "file": "assets/theme-gVQjH5fM.css",
    "integrity": "sha384-tXSaWp9h7f9IDyqcyqxE2oHEre7hVTMpGM1IZPUdwc9do71+G5Fb37fW6KleExPw",
    "isEntry": true,
    "src": "../assets/theme.css",
  },
  "entrypoints/app.css": {
    "file": "assets/app-HiPXl-_u.css",
    "integrity": "sha384-j2SX6yHUu2YQv/067X7HfP4nv2AUV8u5F+we/wI1eQKP0FweNgIiRxCg6vhW6yJE",
    "isEntry": true,
    "src": "entrypoints/app.css",
  },
  "entrypoints/frameworks/vue.js": {
    "css": [
      "assets/vue-Mdlordc1.css",
    ],
    "file": "assets/vue-eqttolrD.js",
    "integrity": "sha384-6gtOC4AW4VZcDnsQqkGMNawlK0PQsbPs0MGQ980Umdx2UHjMHpBXMor4rtgjervR",
    "isEntry": true,
    "src": "entrypoints/frameworks/vue.js",
  },
  "entrypoints/main.ts": {
    "css": [
      "assets/app-HiPXl-_u.css",
      "assets/theme-gVQjH5fM.css",
    ],
    "file": "assets/main-E96L_vrM.js",
    "imports": [
      "entrypoints/frameworks/vue.js",
    ],
    "integrity": "sha384-Vp1tt9FfnLGaJafO0IbRYkaLSCnR2xJj0OIL7Fz040Gx9DwwCZ3CyByPRFy1mBDP",
    "isEntry": true,
    "src": "entrypoints/main.ts",
  },
  "entrypoints/sassy.scss": {
    "file": "assets/sassy-NCcoc604.css",
    "integrity": "sha384-4tkn835gOI7C29uZzlN/PGxd+XyQTG1/9AHpbWYbYLbgWZ1UmDDBFW1iq43MC70L",
    "isEntry": true,
    "src": "entrypoints/sassy.scss",
  },
  "images/logo.png": {
    "file": "assets/logo-Yf1y9V5e.png",
    "integrity": "sha384-zHCg6OQ1T1xX21KMsW9Bb2ECIIMeec41+3KtT0gygdsrmm/Z5qD5TVtTPhyJNGot",
    "src": "images/logo.png",
  },
  "images/logo.svg": {
    "file": "assets/logo-DbqOB_Bf.svg",
    "integrity": "sha384-qQuFPGf9f7DHitjACIjHvY2GwlcFjg+HatTRP3bwG7iAX8MzOW4Sr2dfca6uZlDA",
    "src": "images/logo.svg",
  },
  "index.html": {
    "css": [
      "assets/sassy-NCcoc604.css",
      "assets/app-HiPXl-_u.css",
      "assets/theme-gVQjH5fM.css",
    ],
    "file": "assets/index-ChIRGsHu.js",
    "imports": [
      "entrypoints/main.ts",
      "entrypoints/frameworks/vue.js",
    ],
    "integrity": "sha384-VQYOqTB26w+BvFbNf9g399k8VIDD6LltF/epq+w/O7yKihiZSnV6iGxbK4pAgc02",
    "isEntry": true,
    "src": "index.html",
  },
  "logo.png": {
    "file": "assets/logo-j5j6qipL.png",
    "integrity": "sha384-IGw3tfo4Xt0lp6sO9t5mYtILwfBxDFR8sRCtroxGjhE3r9UxNYgtimkCNzsJU9AP",
    "src": "logo.png",
  },
  "logo.svg": {
    "file": "assets/logo-DbqOB_Bf.svg",
    "integrity": "sha384-qQuFPGf9f7DHitjACIjHvY2GwlcFjg+HatTRP3bwG7iAX8MzOW4Sr2dfca6uZlDA",
    "src": "logo.svg",
  },
}
`;
